daemon/analytics.c

```diff
@@ -335,6 +357,84 @@ void analytics_alarms_notifications(void)
     buffer_free(b);
 }
 
+/*
+ * Checks for the existance of .install_type file and reads it
+ */
+void analytics_get_install_type(void)
+{
+    char *install_type_filename;
+    analytics_set_data_str(&analytics_data.netdata_install_type, "");
+    analytics_set_data_str(&analytics_data.netdata_prebuilt_distro, "");
+
+    install_type_filename =
+        mallocz(sizeof(char) * (strlen(netdata_configured_user_config_dir) + strlen(".install-type") + 2));
+    sprintf(install_type_filename, "%s/%s", netdata_configured_user_config_dir, ".install-type");
+    if (unlikely(access(install_type_filename, R_OK) != 0)) {
+        freez(install_type_filename);
+        return;
+    }
+
+    FILE *fp = fopen(install_type_filename, "r");
+    if (fp) {
+        char *s, *t, buf[256 + 1];
+        size_t len = 0;
+
+        while ((s = fgets_trim_len(buf, 4096, fp, &len))) {
+            if (!strncmp(buf, "INSTALL_TYPE='", 14)) {
+                s = t = buf + 13;
+                if (s) {
+                    while (*s == '\'')
+                        s++;
+                    while (*++t != '\0')
+                        ;
+                    while (--t > s && *t == '\'')
+                        *t = '\0';
+                    if (*s)
+                        analytics_set_data_str(&analytics_data.netdata_install_type, (char *)s);
+                }
+            }
+            else if (!strncmp(buf, "PREBUILT_DISTRO='", 17)) {
+                s = t = buf + 16;
+                if (s) {
```

Comment:
The algorithm inside this `if` and the algorithm inside the previous `if` for `INSTALL_TYPE` has more or less the same code, the exception is one argument for the function `analytics_set_data_str`. I will give a suggestion to convert this for a function.

Reply:
Done with https://github.com/netdata/netdata/pull/11237/commits/49e3c595e0d643e864df710bcc81f801f6768572 !

Reply:
Thank you! :) I will test your PR now.